cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/kafka/Kafka08SimpleApiConsumer.java

```diff
@@ -0,0 +1,408 @@
+import kafka.javaapi.FetchResponse;
+import kafka.javaapi.OffsetRequest;
+import kafka.javaapi.OffsetResponse;
+import kafka.javaapi.PartitionMetadata;
+import kafka.javaapi.TopicMetadata;
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.TopicMetadataResponse;
+import kafka.javaapi.consumer.SimpleConsumer;
+import kafka.message.Message;
+import kafka.message.MessageAndOffset;
+
+import org.apache.twill.kafka.client.BrokerInfo;
+import org.apache.twill.kafka.client.BrokerService;
+import org.apache.twill.kafka.client.TopicPartition;
+import org.apache.twill.zookeeper.RetryStrategies;
+import org.apache.twill.zookeeper.ZKClient;
+import org.apache.twill.zookeeper.ZKClientService;
+import org.apache.twill.zookeeper.ZKClientServices;
+import org.apache.twill.zookeeper.ZKClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * <p>
+ *  The {@link KafkaSimpleApiConsumer} compatible mode for {@code Kafka} 0.8.x releases.
+ *
+ *  The class expect several runtime arguments:
+ *  -) kafka.zookeeper - the location of Kafka Zookeeper
+ *  -) kafka.brokers - comma separate list of Kafka brokers
+ *  -) kafka.partitions - the number of partitions of the topic
+ *  -) kafka.topic - the Kafka topic to get messages for
+ *  -) kafka.default.offset - the default offset for the partition
+ * </p>
+ */
+public class Kafka08SimpleApiConsumer extends KafkaSimpleApiConsumer<String, ByteBuffer, Long>  {
+  private static final Logger LOG = LoggerFactory.getLogger(Kafka08SimpleApiConsumer.class);
+
+  private ZKClientService zkClient;
+  private BrokerService brokerService;
+  private Cache<TopicPartition, SimpleConsumer> kafkaConsumers;
+
+  @Override
+  protected void configureKafka(KafkaConfigurer configurer) {
+    Map<String, String> runtimeArgs = getContext().getRuntimeArguments();
+    if (runtimeArgs.containsKey("kafka.zookeeper")) {
+      configurer.setZooKeeper(runtimeArgs.get("kafka.zookeeper"));
+    } else if (runtimeArgs.containsKey("kafka.brokers")) {
+      configurer.setBrokers(runtimeArgs.get("kafka.brokers"));
+    }
+    setupTopicPartitions(configurer, runtimeArgs);
+  }
+
+  private void setupTopicPartitions(KafkaConsumerConfigurer configurer, Map<String, String> runtimeArgs) {
+    int partitions = Integer.parseInt(runtimeArgs.get("kafka.partitions"));
+    int instanceId = getContext().getInstanceId();
+    int instances = getContext().getInstanceCount();
+    String kafkaTopic = runtimeArgs.get("kafka.topic");
+    for (int i = 0; i < partitions; i++) {
+      if ((i % instances) == instanceId) {
+        configurer.addTopicPartition(kafkaTopic, i);
+      }
+    }
+  }
+
+  @Override
+  protected Iterator<KafkaMessage<Long>> readMessages(KafkaConsumerInfo<Long> consumerInfo) {
```

Comment:
Should be ok because the method is called with iteration to consumer info collections that the value is created with new KafkaConsumerInfo call.